lib/1.4/dml-builtins.dml

```diff
@@ -3588,36 +3595,48 @@ override to have effect.  `write_field` is the interface used for
 access by registers; in most cases, it is easier to express write
 operations using the `write` template.
 
-Note that instantiating `write_field` on a register means that
-register writes behave as if the register consists of one single
-field; a write access will ignore any actual field subobjects in the
-register.  This is often useful in read-only registers, as it allows
-reads to propagate to fields, while a violating write can be handled
-centrally for the whole register.
+Note that instantiating `write_field` on a register incurs a `write_register`
+implementation such that register writes behave as if the register consists
+of one single field; a write access will ignore any actual field subobjects in
+the register. This is often useful in read-only registers, as it allows reads
+to propagate to fields, while a violating write can be handled centrally for
+the whole register.
+
+This `write_register` implementation is provided through the automatically
+instantiated `write_register_via_write_field` template, which you may need
+to reference in order to override the implementation &mdash; for example in
+order to resolve implementation conflicts (through the use of
+[Template-Qualified Method Implementation
+Calls](language.html#template-qualified-method-implementation-calls).
 */
 template write_field {
     shared method write_field(uint64 val, uint64 enabled_bits, void *aux);
 }
 
-template _reg_read_as_field is (register, read_register, read_field) {
+
+template read_register_via_read_field is (register, read_register,
```

Comment:
We may want to polish naming convention a bit. It is correct that write_register is instantiated via write_field, but the result of that is that the write_field method is called via the read_register method, i.e. an instance of "via" with reverse direction.

I haven't thought of a better name yet.

Reply:
trying to map this to standard OO patterns, it looks a bit like Adapter Pattern, but where all three classes are the same. I.e., register needs a read_register, and we have a read_field, so we create an adapter from read_field to read_register.

Would read_field_to_read_register make sense?

Reply:
The main reason for `_to_` would not be length, but  rather familiarity: there is some precedence related to the Adapter pattern (e.g. `LightningToMicroUsbAdapter` in [the wikipedia article](https://en.wikipedia.org/wiki/Adapter_pattern)), whereas I haven't seen `ImplFrom` used for similar patterns elsewhere.

Reply:
Discussed elsewhere, I mixed things up, `read_register` is not a template name so the `_to_` pattern is not applicable.

Reply:
discussed verbally. the templates currently have no good reason to exist but don't really harm anything and Erik is looking into something that _could_ give them a good reason to exist. So: `read_register_from_read_field` is the naming scheme.